shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
```

Comment:
Is there a reason to do this kind of a check at all?

Can do a try around the for loop or something else that would be a better way to support anything that is itterable, or just do nothing (much like the way sort_order is handled).  This kind of argument checking is a little un-pythonic where as long as a class implements the right combo of magic methods you let it pretend to be a list, a dict, or whatever.